lib/dead_end/core_ext.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+# Allow lazy loading, only load code if/when there's a syntax error
+autoload :DeadEnd, "dead_end/api"
+
+# Ruby 3.2+ has a cleaner way to hook into Ruby that doesn't use `require`
+if SyntaxError.new.respond_to?(:detailed_message)
```

Comment:
A new instance will not be needed.
```suggestion
if SyntaxError.method_defined?(:detailed_message)
```

Reply:
Thank you for your PR. I will pull that change in on the next commit.